phases/bitrise_yml.go

```diff
@@ -23,115 +28,163 @@ func currentBranch() (string, error) {
 	return strings.TrimPrefix(out, "refs/heads/"), nil
 }
 
-// BitriseYML ...
-func BitriseYML() (string, string, error) {
-	log.Infof("Setup " + bitriseYMLName)
-	fmt.Println()
-
-	var (
-		err                           error
-		bitriseYMLPath, firstWorkflow string
-	)
-
-	bitriseYMLFile, openErr := os.Open(bitriseYMLName)
-	if openErr != nil && !os.IsNotExist(openErr) {
-		return "", "", fmt.Errorf("failed to open "+bitriseYMLName+", error: %s", openErr)
-	}
-
-	if bitriseYMLFile == nil {
-		log.Warnf("No " + bitriseYMLName + " found in the current working directory.")
-		fmt.Println()
-		const (
-			methodTitle     = "Select method to have a " + bitriseYMLName + ""
-			methodInputPath = "Input path manually"
-			methodScan      = "Generate one by scanning my project"
-		)
-		(&option{
-			title:        methodTitle,
-			valueOptions: []string{methodInputPath, methodScan},
-			action: func(answer string) *option {
-				const (
-					inputTitle = "Enter the path of your " + bitriseYMLName + " file (you can also drag & drop the file here)"
-				)
-				switch answer {
-				case methodInputPath:
-					return &option{
-						title: inputTitle,
-						action: func(answer string) *option {
-							bitriseYMLPath = answer
-							return nil
-						}}
-				case methodScan:
-					branch, branchReadErr := currentBranch()
-					if branchReadErr != nil {
-						err = branchReadErr
-						return nil
-					}
-
-					log.Warnf("The current branch is: %s. Do you want to run the scanner for this branch?", branch)
-					log.Warnf("At this point you can checkout a different branch if you want to.")
-					fmt.Println()
-
-					var scanOption *option
-					scanOption = &option{
-						title: "Hit enter if you have done the checkout or want to use this branch",
-						action: func(answer string) *option {
-							newBranch, branchReadErr := currentBranch()
-							if branchReadErr != nil {
-								err = branchReadErr
-								return nil
-							}
-							if newBranch == branch {
-								cmd := command.New("bitrise", ":init")
-								if err = cmd.SetStderr(os.Stderr).SetStdin(os.Stdin).SetStdout(os.Stdout).Run(); err != nil {
-									err = fmt.Errorf("failed to run command: %s, error: %s", cmd.PrintableCommandArgs(), err)
-									return nil
-								}
-								fmt.Println()
-								bitriseYMLPath = bitriseYMLName
-								return nil
-							}
-
-							branch = newBranch
-
-							log.Warnf("Checked out a different branch: %s. Do you want to run the scanner for this branch?", branch)
-							log.Warnf("At this point you can still checkout a different branch if you want to.")
-							fmt.Println()
-
-							return scanOption
-						}}
-					return scanOption
-				}
-				return nil
-			}}).run()
-	} else {
-		bitriseYMLPath = bitriseYMLName
+func checkBranch(inputReader io.Reader, currentBranch func() (string, error)) error {
```

Comment:
`currentBranch` isn't used anywhere else and neither is `checkBranch` called with a different arg. I'd recommend using a `branch string` parameter instead of the function and passing the return value to `checkBranch`.

Reply:
checkBranch can call currentBranch multiple times.